server/middlewares/token.js

```diff
@@ -0,0 +1,46 @@
+// modules
+const jwt = require('jsonwebtoken');
+
+exports.validate = (req, res, next) => {
+  const token = req.headers['authorization'];
+
+  if (token) {
+    // grab token from headers request
+    const headersToken = token.split(" ")[1];
+
+    jwt.verify(headersToken, "secretKey", err => {
+      if (err) {
+        console.log(err);
+
+        /**
+         * sending response to client
+         * only on if the token is missing
+         */
+        res.status(401).send({
+          status: {
+            code: 401,
+            type: "error"
+          },
+          error: {
+            code: "token_invalid",
+            message: "Token is invalid"
+          }
+        })
+      } else {
+        // move to next middleware on success
+        next();
+      }
+    })
+  } else {
+    res.status(401).send({
+      status: {
+        code: 401,
```

Comment:
If the token is missing, shouldn't it be 400/422?

Reply:
https://httpstatuses.com/401

I am using 401 because it means unauthorised, which means member a missing token is unauthorised.